### ISSUE TITLE
Why the eigenvalue output is different for tensorflow and numpy?

### ISSUE THREAD

**Issue by maye1998** [tensorflow/tensorflow#53222]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version (use command below):
- Python version:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
Why the eigevectors output is different for tensorflow and numpy?
![image](https://user-images.githubusercontent.com/61866921/143771375-8cb7fc00-1482-4507-a963-6635c405497d.png)

**Describe the expected behavior**

**[Contributing](https://www.tensorflow.org/community/contribute)**

- Do you want to contribute a PR? (yes/no):
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by tilakrayal** [comment]
@maye1998 ,
 In order to expedite the trouble-shooting process, could you please provide  code in reproducible format and the TensorFlow version you are using.

**Comment by maye1998** [comment]
> @maye1998 , In order to expedite the trouble-shooting process, could you please provide code in reproducible format and the TensorFlow version you are using.

I'm running this code on Google Colab.
- Tensorflow: 2.7.0
- Numpy: 1.19.5

import numpy as np
import tensorflow as tf

XX = np.random.randn(3, 3)
eig_value1, eig_vector1 = np.linalg.eig(XX)

YY = tf.cast(XX, dtype=tf.float32)
eig_value2, eig_vector2 = tf.linalg.eig(YY)
eig_value1,eig_value2, eig_vector1, eig_vector2

**Comment by tilakrayal** [comment]
@maye1998 ,
Please refer this links [1](https://numpy.org/doc/stable/reference/generated/numpy.linalg.eig.html), [2](https://www.tensorflow.org/api_docs/python/tf/linalg/eig) and SO [link](https://stackoverflow.com/questions/58856160/why-do-tensorflow-and-pytorch-gradients-of-the-eigenvalue-decomposition-differ-f) which provides more information.It helps.Thanks

**Comment by MayeZhang** [comment]
It's different from the problem of the [link](https://stackoverflow.com/questions/58856160/why-do-tensorflow-and-pytorch-gradients-of-the-eigenvalue-decomposition-differ-f)

it's a real random matrix but not a real symmetric matrix, and I want to get the eigenvector of the random matrix.

When I calculate it by numpy and the Tensorflow, we can see that the result of the eigenvalue is same, but the  eigenvector output is different

**Comment by MayeZhang** [comment]
> @maye1998 , Please refer this links [1](https://numpy.org/doc/stable/reference/generated/numpy.linalg.eig.html), [2](https://www.tensorflow.org/api_docs/python/tf/linalg/eig) and SO [link](https://stackoverflow.com/questions/58856160/why-do-tensorflow-and-pytorch-gradients-of-the-eigenvalue-decomposition-differ-f) which provides more information.It helps.Thanks

It's different from the problem of the [link](https://stackoverflow.com/questions/58856160/why-do-tensorflow-and-pytorch-gradients-of-the-eigenvalue-decomposition-differ-f)

it's a real random matrix but not a real symmetric matrix, and I want to get the eigenvector of the random matrix.

When I calculate it by numpy and the Tensorflow, we can see that the result of the eigenvalue is same, but the eigenvector output is different
